syntax = "proto3";

package jogasaki.proto.sql.error;

option java_package = "com.tsurugidb.sql.proto";
option java_outer_classname = "SqlError";

/*
 * Definition of error information for Response.
 */

// the error code for sql execution
enum Code {
    // code not specified
    CODE_UNSPECIFIED = 0;

    // SQL-01000
    SQL_SERVICE_EXCEPTION = 1;

    // SQL-02000
    SQL_EXECUTION_EXCEPTION = 2;

    // SQL-02001
    CONSTRAINT_VIOLATION_EXCEPTION = 3;

    // SQL-02002
    UNIQUE_CONSTRAINT_VIOLATION_EXCEPTION = 4;

    // SQL-02003
    NOT_NULL_CONSTRAINT_VIOLATION_EXCEPTION = 5;

    // SQL-02004
    REFERENTIAL_INTEGRITY_CONSTRAINT_VIOLATION_EXCEPTION = 6;

    // SQL-02005
    CHECK_CONSTRAINT_VIOLATION_EXCEPTION = 7;

    // SQL-02010
    EVALUATION_EXCEPTION = 8;

    // SQL-02011
    VALUE_EVALUATION_EXCEPTION = 9;

    // SQL-02012
    SCALAR_SUBQUERY_EVALUATION_EXCEPTION = 10;

    // SQL-02014
    TARGET_NOT_FOUND_EXCEPTION = 11;

    // SQL-02016
    TARGET_ALREADY_EXISTS_EXCEPTION = 12;

    // SQL-02018
    INCONSISTENT_STATEMENT_EXCEPTION = 13;

    // SQL-02020
    RESTRICTED_OPERATION_EXCEPTION = 14;

    // SQL-02021
    DEPENDENCIES_VIOLATION_EXCEPTION = 15;

    // SQL-02022
    WRITE_OPERATION_BY_RTX_EXCEPTION = 16;

    // SQL-02023
    LTX_WRITE_OPERATION_WITHOUT_WRITE_PRESERVE_EXCEPTION = 17;

    // SQL-02024
    READ_OPERATION_ON_RESTRICTED_READ_AREA_EXCEPTION = 18;

    // SQL-02025
    INACTIVE_TRANSACTION_EXCEPTION = 19;

    // SQL-02027
    PARAMETER_EXCEPTION = 20;

    // SQL-02028
    UNRESOLVED_PLACEHOLDER_EXCEPTION = 21;

    // SQL-02030
    LOAD_FILE_EXCEPTION = 22;

    // SQL-02031
    LOAD_FILE_NOT_FOUND_EXCEPTION = 23;

    // SQL-02032
    LOAD_FILE_FORMAT_EXCEPTION = 24;

    // SQL-02033
    DUMP_FILE_EXCEPTION = 25;

    // SQL-02034
    DUMP_DIRECTORY_INACCESSIBLE_EXCEPTION = 26;

    // SQL-02036
    SQL_LIMIT_REACHED_EXCEPTION = 27;

    // SQL-02037
    TRANSACTION_EXCEEDED_LIMIT_EXCEPTION = 28;

    // SQL-02039
    SQL_REQUEST_TIMEOUT_EXCEPTION = 29;

    // SQL-02041
    DATA_CORRUPTION_EXCEPTION = 30;

    // SQL-02042
    SECONDARY_INDEX_CORRUPTION_EXCEPTION = 31;

    // SQL-02044
    REQUEST_FAILURE_EXCEPTION = 32;

    // SQL-02045
    TRANSACTION_NOT_FOUND_EXCEPTION = 33;

    // SQL-02046
    STATEMENT_NOT_FOUND_EXCEPTION = 34;

    // SQL-02048
    INTERNAL_EXCEPTION = 35;

    // SQL-02050
    UNSUPPORTED_RUNTIME_FEATURE_EXCEPTION = 36;

    // SQL-02052
    BLOCKED_BY_HIGH_PRIORITY_TRANSACTION_EXCEPTION = 37;

    // SQL-02054
    INVALID_RUNTIME_VALUE_EXCEPTION = 38;

    // SQL-02056
    VALUE_OUT_OF_RANGE_EXCEPTION = 39;

    // SQL-02058
    VALUE_TOO_LONG_EXCEPTION = 40;

    // SQL-02060
    INVALID_DECIMAL_VALUE_EXCEPTION = 41;

    reserved 42 to 100;

    // SQL-03000
    COMPILE_EXCEPTION = 101;

    // SQL-03001
    SYNTAX_EXCEPTION = 102;

    // SQL-03002
    ANALYZE_EXCEPTION = 103;

    // SQL-03003
    TYPE_ANALYZE_EXCEPTION = 104;

    // SQL-03004
    SYMBOL_ANALYZE_EXCEPTION = 105;

    // SQL-03005
    VALUE_ANALYZE_EXCEPTION = 106;

    // SQL-03010
    UNSUPPORTED_COMPILER_FEATURE_EXCEPTION = 107;

    reserved 108 to 200;

    // SQL-04000
    CC_EXCEPTION = 201;

    // SQL-04001
    OCC_EXCEPTION = 202;

    // SQL-04010
    OCC_READ_EXCEPTION = 203;

    // SQL-04015
    CONFLICT_ON_WRITE_PRESERVE_EXCEPTION = 204;

    // SQL-04011
    OCC_WRITE_EXCEPTION = 205;

    // SQL-04003
    LTX_EXCEPTION = 206;

    // SQL-04013
    LTX_READ_EXCEPTION = 207;

    // SQL-04014
    LTX_WRITE_EXCEPTION = 208;

    // SQL-04005
    RTX_EXCEPTION = 209;

    // SQL-04007
    BLOCKED_BY_CONCURRENT_OPERATION_EXCEPTION = 210;

}
